tests/acceptance/test_snapshot.py

```diff
@@ -161,33 +164,34 @@ def test_snapshot_avoid_deadlock(self, connection):
     @pytest.mark.only_with_image("uefiimg", "sdimg", "biosimg", "gptimg")
     # Make sure we run both with and without terminal. Many signal bugs lurk in
     # different corners of the console code.
-    @pytest.mark.parametrize(
-        "terminal", ["", "screen -D -m -L -Logfile screen.log.tmp"]
-    )
+    @pytest.mark.parametrize("terminal", ["", "screen -D -m -L -Logfile %%"])
     def test_snapshot_using_mender_artifact(
         self, terminal, bitbake_path, bitbake_variables, connection
     ):
-        try:
+        with tempfile.NamedTemporaryFile(
+            suffix=".mender"
+        ) as artifact, tempfile.NamedTemporaryFile() as screen_log:
+            terminal = terminal.replace("%%", screen_log.name)
             (active, _) = determine_active_passive_part(bitbake_variables, connection)
 
             try:
                 subprocess.check_call(
-                    "%s mender-artifact write rootfs-image %s -n test -t test -o test_snapshot_using_mender_artifact.mender -f ssh://%s@%s:%s"
+                    "%s mender-artifact write rootfs-image %s -n test -t test -o %s -f ssh://%s@%s:%s"
```

Comment:
Tbh, this would've been a prime candidate for an f-string imo

Reply:
I actually started with that, but it became so insanely long that I changed it back! This string is a bit of a catch-22, it's ugly either way, so we can choose between a very long string, or a shorter string with less obvious substitution. In the end I decided to keep it as is, but no strong feelings.

Reply:
`catch-22` strings are the worst!